web/pdf_page_view.js

```diff
@@ -602,6 +606,29 @@ class PDFPageView {
       this._renderXfaLayer();
     }
 
+    // The structure tree is currently only supported when the text layer is
+    // enabled and a canvas is used for rendering.
+    if (this.structTreeLayerFactory && this.textLayer && this.canvas) {
+      // The structure tree must be generated after the text layer for the
+      // aria-owns to work.
+      this._handleTextLayerRendered = event => {
+        if (event.pageNumber !== this.pdfPage.pageNumber) {
+          return;
+        }
+        this.eventBus._off("textlayerrendered", this._handleTextLayerRendered);
+        this.pdfPage.getStructTree().then(tree => {
+          const treeDom = this.structTreeLayer.render(tree);
+          this.structTree = treeDom;
+          treeDom.classList.add("structTree");
+          this.canvas.appendChild(treeDom);
+        });
+      };
+      this.eventBus._on("textlayerrendered", this._handleTextLayerRendered);
```

Comment:
There's no guarantee that this event listener is ever removed, if the page is e.g. destroyed, hence please add something like the following [to the `cancelRendering`-method](https://github.com/mozilla/pdf.js/blob/ca7f546828603d15ac0975f6131669321bfccceb/web/pdf_page_view.js#L359):
```javascript
if (this._onTextLayerRendered) {
  this.eventBus._off("textlayerrendered", this._onTextLayerRendered);
  this._onTextLayerRendered = null;
}
```